session.rst

```diff
@@ -1,15 +1,276 @@
+.. index::
+   single: HTTP
+   single: HttpFoundation, Sessions
+
 Sessions
 ========
 
-Symfony provides a session object and several utilities that you can use to
-store information about the user between requests.
+Installation
+------------
+
+The Symfony HttpFoundation component has a very powerful and flexible session
+subsystem which is designed to provide session management that you can use to
+store information about the user between requests through a clear
+object-oriented interface using a variety of session storage drivers.
+
+You need to install HttpFoundation to handle sessions:
+
+.. code-block:: terminal
+
+    $ composer require symfony/http-foundation
+
+Symfony sessions are designed to replace the usage of the ``$_SESSION`` super
+global and native PHP functions related to manipulating the session like
+``session_start()``, ``session_regenerate_id()``, ``session_id()``,
+``session_name()``, and ``session_destroy()``.
+
+.. note::
+
+    Sessions are only started if you read or write from it.
+
+.. _session-intro:
+
+Basic Usage
+-----------
+
+The session is available through the ``Request`` object and the ``RequestStack``
+service. Symfony injects the ``request_stack`` service in services and controllers
+if you type-hint an argument with :class:`Symfony\\Component\\HttpFoundation\\RequestStack`::
+
+.. configuration-block::
+
+    .. code-block:: php-symfony
+
+        use Symfony\Component\HttpFoundation\RequestStack;
+
+        class SomeService
+        {
+            private $requestStack;
+
+            public function __construct(RequestStack $requestStack)
+            {
+                $this->requestStack = $requestStack;
+
+                // Accessing the session in the constructor is *NOT* recommended, since
```

Comment:
Imho this should be a note and not a code example if it is not recommended

Reply:
That's something that I copy/pasted, so not new. I would keep it like this as this is about basic usage, and this one is bad.